### PR TITLE
[CI] Enable Multibranch Pipeline with JUnit transformation

### PR DESCRIPTION
**Pull request by v1v**
### What

- Enable Multibranch Pipeline with JUnit transformation.
- Run for linux, macos

### Issues

Blocked by https://github.com/elastic/gmux/pull/2

### Follow ups

If required we can enable windows, so far, it's not yet enabled since apparently `make` was not available at the moment, and there was not really a specific requirement to be supported though.

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      matrix {
+        options { skipDefaultCheckout() }
+        axes {
+          axis {
+            name 'GO_VERSION'
+            values '1.15.6'
+          }
+          axis {
+            name 'PLATFORM'
+            values 'ubuntu-18 && immutable', 'macosx', 'windows-2019 && immutable'
```
Comment (v1v) [inline .ci/Jenkinsfile:46]:
```suggestion
            values 'ubuntu-18 && immutable', 'macosx', 'windows-2019-immutable'
```

**Review by cachedout — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -21,4 +21,4 @@ fmt:
 
 .PHONY: test
 test:
-	go test -race -v ./...
+	go run gotest.tools/gotestsum --format testname -- -race -v ./...
```
Comment (kuisathaverat) [inline Makefile:24]: it does not generate JUnit report, Is it intended? if not

```suggestion
	go run gotest.tools/gotestsum --format testname --junitfile junit-report.xml -- -race -v ./...
```

**Review by kuisathaverat — APPROVED** [review]
LGTM

**Pull request merged**